dmutils/forms/mixins.py

```diff
@@ -56,15 +56,26 @@ class DMFieldMixin:
     Derived classes which include this mixin should have a
     subclass of `wtforms.Field` in their base classes.
     '''
-    def __init__(self, label=None, validators=None, hint=None, **kwargs):
+    def __init__(self, label=None, validators=None, hint=None, question_advice=None, **kwargs):
         super().__init__(label=label, validators=validators, **kwargs)
-        self.hint = hint or getattr(self.__class__, 'hint', None)
+        if hint:
+            self.hint = hint
+        if question_advice:
+            self.question_advice = question_advice
+
+        # wtforms.Field overwrites self.type on init
+        # if we want to specify it on a subclass
+        # this line will bring it back
```

Comment:
Ah iiiiinteresting. :+1:

(is the attribute name `type` already determined for us or do we get a choice? calling things `type` is a pet peeve of mine, and not just because it leads to people inadvertently redefining `type()`...)

Reply:
Yeah, we want there to be an attribute called `type` because some of the form templates expect a variable with that name.

Normally WTForms saves the class name to `type`, which is a pain if you want it to be something else.

Reply:
:grumble: ah well.